### PR TITLE
[Merged by Bors] - feat(Order/WithBot): Equiv.withBotSubtypeNe

### PR DESCRIPTION
**Pull request by nomeata**
Needed in #19210

From the Carleson project

### CODE REVIEW

**Comment by nomeata** [comment]
Questions for the review:

* What should the name be?
* Is it correct to add `@[simps]` here?

File: Mathlib/Order/WithBot.lean [hunk 1]
```diff
@@ -158,6 +158,13 @@ theorem eq_unbot_iff {a : α} {b : WithBot α} (h : b ≠ ⊥) :
   · simpa using h rfl
   · simp
 
+/-- The equivalence between the non-bottom elements of `WithBot α` and `α`. -/
+def subType_ne_bot_equiv : {y : WithBot α // y ≠ ⊥} ≃ α where
```
Comment (nomeata) [inline Mathlib/Order/WithBot.lean:162]:
```suggestion
@[simps] def subType_ne_bot_equiv : {y : WithBot α // y ≠ ⊥} ≃ α where
```

**Review by YaelDillies — COMMENTED** [review]
The `Option` version is called `Equiv.optionSubtypeNe`

**Comment by nomeata** [comment]
Not quite the analogue statement, but good enough

**Comment by YaelDillies** [comment]
Ah then the actually analogous statement should be used in some categorical construction around `Pointed` to show that the category of pointed types with pointed maps is equivalent to the category of types with partial maps

**Comment by nomeata** [comment]
You lost me somewhere :-). Should I import category theory into `WithBot`? 😈

**Comment by Ruben-VandeVelde** [comment]
Please fix the pr title

**Review by YaelDillies — APPROVED** [review]
maintainer delegate

otherwise, thanks!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**